### PR TITLE
Use plain path.join to fix path handling

### PR DESCRIPTION
**Pull request by develar**
Fix error "Path to copy doesn't exist D:\dev\projects\privateName\src\electron\dev\projects\ privateName\src\electron\node_modules\electron-winstaller\vendor\Update.exe"

use plain path.join — path helper doesn't add any value

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -75,21 +75,21 @@ export async function createWindowsInstaller(options) {
   let { appDirectory, outputDirectory, loadingGif } = options;
   outputDirectory = p`${outputDirectory || 'installer'}`;
 
-  await jetpack.copyAsync(
-    p`${__dirname}/../vendor/Update.exe`,
-    p`${appDirectory}/Update.exe`);
+  const vendorPath = path.join(__dirname, "..", "vendor");
```
Comment (havenchyk) [inline src/index.js]: `path.resolve('../vendor')` can be used instead.

File: src/index.js [hunk 2]
```diff
@@ -75,21 +75,21 @@ export async function createWindowsInstaller(options) {
...
+  await jetpack.copyAsync(path.join(vendorPath, "Update.exe"), path.join(appDirectory, "Update.exe"));
 
-  let defaultLoadingGif = p`${__dirname}/../resources/install-spinner.gif`;
+  let defaultLoadingGif = path.join(__dirname, "..", "resources", "install-spinner.gif");
```
Comment (havenchyk) [inline src/index.js]: Same. `path.resolve("../resources/install-spinner.gif");` I believe it works on windows. at least in REPL

**Comment by lukeapage** [comment]
This probably fixes #13. You can also use jetpack more for path resolution.

File: src/index.js (hunk 1 above)
Comment (develar) [inline src/index.js]: No. No. See docs — "the current working directory is used as well".

File: src/index.js (hunk 2 above)
Comment (develar) [inline src/index.js]: No. No. See docs — "the current working directory is used as well".

**Comment by develar** [comment]
@lukeapage Ci failed due to unrelated error. Could you please restart AppVeyor build? Otherwise I have to push some dummy change to trigger rebuild.

**Comment by develar** [comment]
@lukeapage I want to get rid of jetpack because jetpack uses q instead of bluebird. [Bluebird ships with the best cross-platform long stack traces](http://bluebirdjs.com/docs/why-bluebird.html). fs-extra + Promise.promisifyAll is enough. No need yet another fs lib (electron-builder and electron-packager uses fs-extra).

**Comment by lukeapage** [comment]
@develar I'm not a admin on this repo, I just fixed a few things recently and took a look at the PR - so I can't restart appveyor - the same thing is happening on my PR's where I've just changed documentation. You can't see who is an admin on electronjs, the only person I know for sure is admin is @kevinsawicki but he is usually quite fast at looking at PR's

re: jetpack - fine by me

**Comment by paulcbetts** [comment]
I don't think this is related to path.join, I think this is a jetpack issue. I had to replace several methods with `fs` versions and they would work, it appears that jetpack will insert `./`'s into paths which throws them off

**Comment by develar** [comment]
@paulcbetts path helper does `path.resolve(...parts)` and on error `path.join(...parts);` For what? Why not use just `path.join`? Yes, I want to get rid of jetpack and use promisified fs-extra. What do you think? I can prepare PR.

**Comment by RiderSx** [comment]
@paulcbetts replacing path-helper with path.join solves the problem, tested it

**Comment by develar** [comment]
@RiderSx +1 electron-builder was published (2.8.8) with my changes and user reported that issue is gone.

**Comment by develar** [comment]
And yes — don't use windows-installer yet, use only previous version (not es6). A lot of errors — at least  https://github.com/loopline-systems/electron-builder/issues/208 electron-builder 2.8.6 will continue to use my own fixed version.

**Comment by havenchyk** [comment]
I can confirm that still have errors during building on windows. Even after ths PR applying.

**Comment by havenchyk** [comment]
@kevinsawicki could you please take a look?

**Comment by develar** [comment]
PR updated — path-helper.js completely removed.

**Comment by develar** [comment]
Confirmed by @havenchyk that `path-helper.js completely removed` fixes the issue https://github.com/loopline-systems/electron-builder/issues/208 I hope it will be merged, we cannot yet use original winstaller, but finally I hope all my changes will be accepted.

**Comment by kevinsawicki** [comment]
Minor style comment, but it looks like both single quotes and double quotes are used in the new `path.join` calls.

It would be nice to be consistent and maybe just standardize on single quotes.

**Comment by develar** [comment]
@kevinsawicki Fixed, eslint rule added to avoid such error in the future.

**Comment by kevinsawicki** [comment]
>  eslint rule added to avoid such error in the future.

Awesome, thanks :+1:

**Pull request merged**